### ISSUE TITLE
Return instrumentation can be at an unexpected place ...

### ISSUE THREAD

**Issue by lks9** [lks9/src-tracer#16]
... when we have a function call in the return statement like this

```C
int foo(void) {
    return 0;
}
int main(int argc, char **argv) {
    return foo();
}
```

then we actually get the trace
```
F2 main
R
F1 foo
R
```
instead of:
```
F2 main
|  F1 foo
|  R
R
```

The reason is that the instrumentation currently produces
```C
  _RETURN
  return foo();
```
instead of:
```C
  auto result = foo(); 
  _RETURN
  return result;
```

We will not change the instrumentation because
1. it would be an unnecessary refactoring step and
2. it would remove the tail recursion in the instrumented code. This can affect the performance of the instrumented code. In trace mode, the performance is crucial.

**Comment by lks9** [comment]
Maybe there is still some hope that this one could be solved after #45.

**Comment by lks9** [comment]
Removed as duplate of #45.
